nanoCAT/recipes/coordination_number.py

```diff
@@ -0,0 +1,57 @@
+import numpy as np
+from itertools import combinations
+from scm.plams import Molecule
+from FOX import group_by_values
+from nanoCAT.bde.guess_core_dist import guess_core_core_dist
+
+mol = Molecule('perovskite.xyz')
+d_outer = 5.1 
+
+xyz = np.asarray(mol)
+
+# Construct a dictionary with atomic symbols as keys and arrays of indices as values
+elements = [at.symbol for at in mol.atoms]
+symbol_enumerator = enumerate(elements)
+idx_dict = group_by_values(symbol_enumerator)
+for k, v in idx_dict.items():
+    idx_dict[k] = np.fromiter(v, dtype=int, count=len(v))
+    
+# Construct two arrays the indice-pairs of all possible combinations in idx_dict.
+# The combinations, by definition, do not contain any atom pairs where ``at1.symbol == at2.symbol``
+x, y = [], []
```

Comment:
The construction of ``x`` and ``y`` would be nice to put in its own function, as it is rather self contained and well-defined in scope.

This is some quite a general advice actually, think about how the script can be (logically) divided into smaller functions.